src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/ComputableGraphStructure.java

```diff
@@ -0,0 +1,286 @@
+        evalStrategyForAllNodes.sort(Comparator.comparingInt(depthsMap::get));
+
+        /* evaluation strategy for updating the descendents of a mutated node */
+        evalStrategyForNodeMutation = new HashMap<>();
+        for (final String mutNode : nodeKeysSet) {
+            final Set<String> allInvolvedSet = new HashSet<>();
+            allInvolvedSet.add(mutNode);
+            for (final String desc : allDescendentsMap.get(mutNode)) {
+                allInvolvedSet.add(desc);
+                allInvolvedSet.addAll(allParentsMap.get(desc));
+            }
+            final List<String> allInvolvedList = new ArrayList<>();
+            allInvolvedList.addAll(allInvolvedSet);
+            /* sort by depth */
+            allInvolvedList.sort(Comparator.comparingInt(depthsMap::get));
+            evalStrategyForNodeMutation.put(mutNode, allInvolvedList);
+        }
+    }
+
+    /**
+     * Updates the depth of a node recursively
+     *
+     * @param nodeKey the key of the node to update
+     */
+    private void updateDepth(final String nodeKey) {
+        if (depthsMap.get(nodeKey) == null) {
+            immediateParentsMap.get(nodeKey).forEach(this::updateDepth);
+            final int depth = Collections.max(immediateParentsMap.get(nodeKey).stream().map(depthsMap::get)
+                    .collect(Collectors.toList())) + 1;
+            if (depth > nodeKeysSet.size() - 1) {
+                throw new CyclicGraphException("The graph has cycles");
+            }
+            depthsMap.replace(nodeKey, depth);
+        }
+    }
+
+    public Set<String> getNodeKeysSet() { return nodeKeysSet; }
+
+    public Set<String> getNodeTagsSet() { return nodeTagsSet; }
+
+    public Map<String, Set<String>> getAllTagsMap() { return allTagsMap; }
+
+    public Map<String, Set<String>> getImmediateDescendentsMap() { return immediateDescendentsMap; }
+
+    public Map<String, Set<String>> getImmediateParentsMap() { return immediateParentsMap; }
+
+    public Map<String, Set<String>> getAllDescendentsMap() { return allDescendentsMap; }
+
+    public Map<String, Set<String>> getAllParentsMap() { return allParentsMap; }
+
+    public Map<String, Integer> getDepthsMap() { return depthsMap; }
+
+    public Map<Integer, Set<String>> getNodesByDepthMap() { return nodesByDepthMap; }
+
+    public Map<String, Set<String>> getNodesByTagMap() { return  nodesByTagMap; }
+
+    public List<String> getEvalStrategyByEvaluatedNode(final String nodeKey) {
+        return evalStrategyForNodeEvaluation.get(nodeKey);
+    }
+
+    public List<String> getEvalStrategyByMutatedNode(final String nodeKey) {
+        return evalStrategyForNodeMutation.get(nodeKey);
+    }
+
+    public List<String> getEvalStrategyByTag(final String tagKey) {
+        return evalStrategyForTagEvaluation.get(tagKey);
+    }
+
+    public List<String> getEvalStrategyForAllNodes() {
+        return evalStrategyForAllNodes;
+    }
+
+    public String statusToString() {
```

Comment:
changed it to `@Override public String toString()`. It is useful for IntelliJ debugging.